src/clients/entitlements/entitlements.user-scoped.ts

```diff
@@ -95,21 +131,30 @@ export class EntitlementsUserScoped<T extends TEntity = TEntity> {
     };
   }
 
-  isEntitledTo(featureOrPermission: IsEntitledToPermissionInput): Promise<IsEntitledResult>;
-  isEntitledTo(featureOrPermission: IsEntitledToFeatureInput): Promise<IsEntitledResult>;
-  async isEntitledTo({
-    featureKey,
-    permissionKey,
-  }: {
-    permissionKey?: string;
-    featureKey?: string;
-  }): Promise<IsEntitledResult> {
+  isEntitledTo(
+    featureOrPermission: IsEntitledToPermissionInput,
+    attributes?: Record<string, any>,
```

Comment:
We don't support nested objects in the attributes,
so the attributes type should be more closed 

for example, this is what I defined in the common package. you can use it.
`export type CustomAttributes = Record<string, string | number | boolean | Date>;`

Reply:
the problem is that `CustomAttributes` type and return type of `prepareAttributes` function don't match

Reply:
why is it a problem? since this function signature is exposed only for the user

Reply:
I think I made my way around it